### PR TITLE
Fix Dart 2 runtime issues.

### PR DESCRIPTION
**Pull request by jakobr-google**
Upgrade to Dart 2.0.0-dev.54.0 to fix issues with mirrors in Dart 2
mode. Added lots of types.

### CODE REVIEW

File: lib/src/datastore_impl.dart [hunk 1]
```diff
@@ -267,7 +267,7 @@ class DatastoreImpl implements datastore.Datastore {
     return orders.map(_convertDatastore2ApiOrder).toList();
   }
 
-  static Future _handleError(error, stack) {
+  static FutureOr<Null> _handleError(error, stack) {
```
Comment (mkustermann) [inline lib/src/datastore_impl.dart]: Why `FutureOr`? This function is guaranteed to return a `Future`?

File: lib/src/db/annotations.dart [hunk 2]
```diff
@@ -298,4 +298,11 @@ class StringListProperty extends ListProperty {
   const StringListProperty({String propertyName, bool indexed: true})
       : super(const StringProperty(),
             propertyName: propertyName, indexed: indexed);
+
+  @override
+  Object decodePrimitiveValue(ModelDB db, Object value) {
+    return (super.decodePrimitiveValue(db, value) as core.List).cast<String>();
+  }
+
+
```
Comment (mkustermann) [inline lib/src/db/annotations.dart]: Maybe remove extra lines here.

File: lib/src/datastore_impl.dart (hunk 1 above)
Comment (jakobr-google) [reply lib/src/datastore_impl.dart]: Yeah. The `<Null>` was the important bit. Fixed.

File: lib/src/db/annotations.dart (hunk 2 above)
Comment (jakobr-google) [reply lib/src/db/annotations.dart]: Done.

**Pull request merged**